sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/Metadata.scala

```diff
@@ -0,0 +1,304 @@
+   * added.
+   *
+   * @param key
+   *   The key to add.
+   * @param value
+   *   The value to add.
+   * @return
+   *   A copy of this Metadata object with the entry added.
+   */
+  def add(key: String, value: String): Metadata
+
+  /**
+   * Add the binary value for the given key.
+   *
+   * <p>This will not replace any existing entries, it will simply append the entry to the end of the list.
+   *
+   * <p>This method does not modify this Metadata object, it returns a copy of this Metadata object with the entry
+   * added.
+   *
+   * @param key
+   *   The key to add.
+   * @param value
+   *   The value to add.
+   * @return
+   *   A copy of this Metadata object with the entry added.
+   */
+  def addBinary(key: String, value: ByteBuffer): Metadata
+
+  /**
+   * Remove all metadata entries with the given key.
+   *
+   * <p>The key will be matched against entries case insensitively.
+   *
+   * <p>This method does not modify this Metadata object, it returns a copy of this Metadata object with the entries
+   * removed.
+   *
+   * @param key
+   *   The key to remove.
+   * @return
+   *   A copy of this Metadata object with the entries removed.
+   */
+  def remove(key: String): Metadata
+
+  /**
+   * Clear all metadata entries.
+   *
+   * <p>This method does not modify this Metadata object, it returns an empty Metadata object.
+   *
+   * @return
+   *   An empty metadata object.
+   */
+  def clear(): Metadata
+
+  /**
+   * Whether this metadata is also a CloudEvent.
+   *
+   * <p>This will return true if all of the required CloudEvent fields are set, that is, the specversion, id, source and
+   * type fields.
+   *
+   * @return
+   *   True if the CloudEvent required attributes are set in this Metadata.
+   */
+  def isCloudEvent: Boolean
+
+  /**
+   * Return a CloudEvent representation of this Metadata.
+   *
+   * <p>Note that the CloudEvent representation will retain any non CloudEvent metadata when converted back to Metadata.
+   *
+   * @return
+   *   This Metadata expressed as CloudEvent metadata.
+   * @throws IllegalStateException
+   *   If this metadata is not a CloudEvent, that is, if it doesn't have any of specversion, id, source or type
+   *   CloudEvent fields defined.
+   */
+  def asCloudEvent: CloudEvent
+
+  /**
+   * Convert this metadata to a CloudEvent, adding the given required CloudEvent fields.
+   *
+   * <p>Any metadata in this Metadata object will be left intact when asMetadata is called
+   *
+   * @param id
+   *   The id of the CloudEvent.
+   * @param source
+   *   The source of the CloudEvent.
+   * @param type
+   *   The type of the CloudEvent.
+   * @return
+   *   This metadata, represented as a CloudEvent with the specified fields.
+   */
+  def asCloudEvent(id: String, source: URI, `type`: String): CloudEvent
+}
+
+object Metadata {
+  def empty = new MetadataImpl(JMetadataImpl.Empty)
```

Comment:
always define explicit return types in public api: `empty: Metadata`

Reply:
ah yes, I forget this every time!